gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/io/binary/types/EdgeSerializer.java

```diff
@@ -42,12 +44,15 @@ public EdgeSerializer() {
     @Override
     protected Edge readValue(final Buffer buffer, final GraphBinaryReader context) throws IOException {
         final Object id = context.read(buffer);
-        final String label = context.readValue(buffer, String.class, false);
+        // reading single string value for now according to GraphBinaryV4
+        final String label = (String) context.readValue(buffer, ArrayList.class, false).get(0);
```

Comment:
Shouldn't this just be "List.class" instead of "ArrayList.class" since the default typeregistry registers List not ArrayList and List is more general?